streams/src/main/java/org/apache/kafka/streams/processor/internals/StandbyTask.java

```diff
@@ -179,11 +181,12 @@ private void close(final boolean clean) {
                 transitionTo(State.CLOSING);
             }
 
-            if (state() == State.CLOSING) {
-                executeAndMaybeSwallow(clean, () -> {
-                    StateManagerUtil.closeStateManager(log, logPrefix, clean,
-                        false, stateMgr, stateDirectory, TaskType.STANDBY);
-                }, "state manager close", log);
+            if (state() == State.CLOSING || state() == State.RESTORING) {
```

Comment:
fix 1

Reply:
Standby task should never be in RESTORING since we always transit from CREATED -> RUNNING -> RESTORING in one call. Did you observe this was not the case from failed system tests? Even in unclean close case you described I did not see why it could be possible..